lib/neo4j/active_node/has_n.rb

```diff
@@ -183,7 +168,16 @@ def #{name}(node = nil, rel = nil, query_proxy = nil)
             #{name}_query_proxy(query_proxy: query_proxy, node: node, rel: rel, context: context)
           end}, __FILE__, __LINE__)
       end
+
+      private
+
+      def build_association(macro, direction, name, options)
+        association = Neo4j::ActiveNode::HasN::Association.new(macro, direction, name, options)
+        @associations ||= {}
+        @associations[name] = association
+        create_reflection(macro, name, association, self)
+        association
```

Comment:
I have this semi-weird style thing where I like to use `#tap` for this sort of thing.  What do you think?

    Neo4j::ActiveNode::HasN::Association.new(macro, direction, name, options).tap do |association|
      @associations ||= {}
      @associations[name] = association
      create_reflection(macro, name, association, self)
    end